src/main/scala/beam/router/r5/RoutingWorker.scala

```diff
@@ -53,7 +66,7 @@ class R5RoutingWorker(workerParams: WorkerParameters) extends Actor with ActorLo
 
   private val tickTask: Cancellable =
     context.system.scheduler.scheduleWithFixedDelay(2.seconds, 10.seconds, self, "tick")(context.dispatcher)
-  private var msgs: Long = 0
+  private val msgs = new AtomicLong()
```

Comment:
Actor executes the received messages in single threaded mode (until you create Future or Thread manually). So it is not needed to have it as Atomic

Reply:
makes sense, for us it was previously used inside of future, so i've changed it to atomic
reverted it back now